vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SpaceManagerTest.java

```diff
@@ -116,23 +127,38 @@ public void test_storeSpace_success() throws FileStorageException {
         Mockito.verify(spaceDisplayRepo).save((SpaceDisplay)spaceDisplay);
     }
 
-
-
-    @Test(expected = SpaceDoesNotExistException.class)
-    public void test_deleteSpaceById_whenIdIsNull() throws SpaceDoesNotExistException {
-        Mockito.doThrow(IllegalArgumentException.class)
-        .when(spaceRepo).deleteById(null);
+    @Test
+    public void test_deleteSpaceById_whenIdIsNull() throws SpaceDoesNotExistException{
         managerToTest.deleteSpaceById(null);
+        Mockito.verify(spaceDisplayRepo, Mockito.never()).deleteBySpaceId(null);
+        Mockito.verify(spaceRepo, Mockito.never()).deleteById(null);
     }
 
-    @Test(expected = SpaceDoesNotExistException.class)
-    public void test_deleteSpaceById_forNonExistentId() throws SpaceDoesNotExistException {
-        String imgId = "imgId";
-        Mockito.doThrow(EmptyResultDataAccessException.class)
-        .when(spaceRepo).deleteById(imgId);
-        managerToTest.deleteSpaceById(imgId);
+    @Test
+    public void test_deleteSpaceById_forNonExistentId() {  
+        managerToTest.deleteSpaceById(spaceId);
+        Mockito.verify(spaceDisplayRepo).deleteBySpaceId(spaceId);
+        Mockito.verify(spaceRepo).deleteById(spaceId);
     }
 
+    @Test
+    public void test_deleteSpaceById_forSuccess() {  
+        managerToTest.deleteSpaceById(spaceId1);
+        Mockito.verify(spaceDisplayRepo).deleteBySpaceId(spaceId1);
+        Mockito.verify(spaceRepo).deleteById(spaceId1);
+    }
+    
+    @Test
+    public void test_deleteSpaceById_whenSpaceHasLinks() {  
+        spaceLink = new SpaceLink();
+        spaceLink.setId(spaceId2);
+        Mockito.when(spaceLinkRepo.getLinkedSpaces(spaceId1)).thenReturn(Arrays.asList(spaceLink));
+        managerToTest.deleteSpaceById(spaceId1);
+        Mockito.verify(spaceLinkDisplayRepo).deleteBySpaceLinkId(spaceId2);
+        Mockito.verify(spaceDisplayRepo).deleteBySpaceId(spaceId1);
+        Mockito.verify(spaceRepo).deleteById(spaceId1);
```

Comment:
what about the links? you need to make sure they are being deleted.

Reply:
and what about incoming links?

Reply:
For the outgoing links, I am checking if the links are being deleted at line 157.

Reply:
157 checks with the linkDisplayRepo not the linkrepo, does it?